src/openrct2/actions/LandSetHeightAction.cpp

```diff
@@ -33,6 +33,13 @@ LandSetHeightAction::LandSetHeightAction(const CoordsXY& coords, uint8_t height,
 {
 }
 
+void LandSetHeightAction::AcceptParameters(GameActionParameterVisitor& visitor)
```

Comment:
Is there any need for plugins to have access to both the set height and the Raise/Lower actions. You would know better than I would on if they would be useful.

Reply:
The Raise/Lower actions include the logic for "pulling a tile up" by a corner.
In the SetHeight action you need to define for yourself how the surface should look like after the operation.

Reply:
so do you need the setheight action?

Reply:
Yes, because you can define the surface shape.

"I want to set this surface to height Z and have the northwest corner up." -> one setheight call, potentially `|oldZ-newZ|` many raise/lower calls.

"I want to raise the northwest corner of this surface by one." -> one simple raise call, one complicated setheight call that needs to implement a big chunk of the raise logic.

Reply:
okay then.